lib/github_changelog_generator/gitlab_fetcher.rb

```diff
@@ -0,0 +1,449 @@
+
+      # separate arrays of issues and pull requests:
+      [issues.map { |issue| stringify_keys_deep(issue.to_hash) }, fetch_closed_pull_requests]
+    end
+
+    # Fetch all pull requests. We need them to detect :merged_at parameter
+    #
+    # @return [Array <Hash>] all pull requests
+    def fetch_closed_pull_requests
+      pull_requests = []
+      options = { state: "merged", scope: :all }
+
+      @client.merge_requests(@project_id, options).auto_paginate do |new_pr|
+        new_pr = stringify_keys_deep(new_pr.to_hash)
+        # align with Github naming
+        new_pr["number"] = new_pr["iid"]
+        new_pr["html_url"] = new_pr["web_url"]
+        new_pr["merged_at"] = new_pr["updated_at"]
+        new_pr["pull_request"] = true
+        new_pr["user"] = { login: new_pr["author"]["username"], html_url: new_pr["author"]["web_url"] }
+        # to make it work with older gitlab version or repos that lived across versions
+        new_pr["merge_commit_sha"] = new_pr["merge_commit_sha"].nil? ? new_pr["sha"]: new_pr["merge_commit_sha"]
+        pull_requests.push(new_pr)
+      end
+
+      print_empty_line
+
+      Helper.log.info "Pull Request count: #{pull_requests.count}"
+      pull_requests.map { |pull_request| stringify_keys_deep(pull_request.to_hash) }
+    end
+
+    # Fetch event for all issues and add them to 'events'
+    #
+    # @param [Array] issues
+    # @return [Void]
+    def fetch_events_async(issues)
+      i       = 0
+      threads = []
+      options = { }
+      if issues.empty?
+        return
+      end
+      options[:target_type] = issues.first["merged_at"].nil? ? "issue" : "merge_request"
+      issue_events = []
+      @client.project_events(@project_id, options).auto_paginate do |event|
+        event = stringify_keys_deep(event.to_hash)
+        # gitlab to github
+        event["event"] = event["action_name"]
+        issue_events.push(event)
+      end
+      # p issue_events
+
+      issues.each_slice(MAX_THREAD_NUMBER) do |issues_slice|
+        issues_slice.each do |issue|
+          threads << Thread.new do
+            issue["events"] = []
+            issue_events.each do |new_event|
+              if issue["id"] == new_event["target_id"]
+                if new_event["action_name"].eql? "closed"
+                  issue["closed_at"] = issue["closed_at"].nil? ? new_event["created_at"] : issue["closed_at"]
+                end
+                issue["events"].push(new_event)
+              end
+            end
+            print_in_same_line("Fetching events for #{options[:target_type]}s: #{i + 1}/#{issues.count}")
+            i += 1
+          end
+        end
+        threads.each(&:join)
+        threads = []
+      end
+
+      # to clear line from prev print
+      print_empty_line
+
+      Helper.log.info "Fetching events for issues and PR: #{i}"
+    end
+
+    # Fetch comments for PRs and add them to "comments"
+    #
+    # @param [Array] prs The array of PRs.
```

Comment:
`    # @param [Array] prs The array of PRs.`

could be written more correctly with the name of the parameter first. (The type is clearly shown in the YARD documentation, so we leave that out of the descriptive note.)

`    # @param prs [Array] PRs for which to fetch comments`